### PR TITLE
chore(package): upgrade to Babel 7

### LINKED ISSUE #1900: Babel's stage presets are deprecated
https://babeljs.io/blog/2018/07/27/removing-babels-stage-presets

We will probably want to migrate before upgrading to future releases.

### PR DESCRIPTION
**Pull request by saschanaz**
Easy enough and fun, especially [`"targets": "defaults"`](https://babeljs.io/docs/en/babel-preset-env#targets).

Fixes #1900

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -1,6 +1,11 @@
 {
   "presets": [
-    "babel-preset-stage-3-without-async"
-  ],
-  "plugins": ["transform-es2015-modules-amd"]
+    [
+      "@babel/preset-env",
+      {
+        "targets": "defaults",
+        "modules": "amd"
```
Comment (marcoscaceres) [inline .babelrc]: Nice that they added this by default too.

**Comment by saschanaz** [comment]
Oops, looks like I missed something.

**Comment by marcoscaceres** [comment]
Argh... `regeneratorRuntime` means it's trying to find the Bable runtime :( Definitely don't want that. Which browsers does "default" exactly represent?

**Comment by saschanaz** [comment]
`defaults` means [`> 0.5%, last 2 versions, Firefox ESR, not dead`](https://github.com/browserslist/browserslist#full-list).

**Comment by marcoscaceres** [comment]
> defaults: Browserslist’s default browsers (> 0.5%, last 2 versions, Firefox ESR, not dead).

I think that's too wide... tbh.

**Comment by marcoscaceres** [comment]
Try: "last 1 versions"?

**Comment by saschanaz** [comment]
@marcoscaceres I found https://browserl.ist/ to see what's the exact result and learned that `defaults` includes IE 🤣

**Comment by marcoscaceres** [comment]
Lol, poor IE. I miss it :)

**Comment by saschanaz** [comment]
I cannot exclude the old-spec browsers easily, so I instead explicitly picked the latest versions of Firefox/Chrome/Safari/Edge. I think it's what we want, anyway.

File: .babelrc [hunk 2]
```diff
@@ -1,6 +1,11 @@
 {
   "presets": [
-    "babel-preset-stage-3-without-async"
-  ],
-  "plugins": ["transform-es2015-modules-amd"]
+    [
+      "@babel/preset-env",
+      {
+        "targets": "last 1 Firefox version, last 1 Chrome version, last 1 Safari version, last 1 Edge version",
```
Comment (marcoscaceres) [inline .babelrc]: Yeah, that looks good!

**Comment by saschanaz** [comment]
👍

**Pull request merged**